src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java

```diff
@@ -127,14 +128,24 @@ private static void createVoltageControls(List<LfBus> lfBuses, LfNetworkParamete
                     }
                 });
 
-                if (parameters.isGeneratorVoltageRemoteControl() || controlledBus == controllerBus) {
+                boolean keepVoltageRemoteControl = false;
+                if (!parameters.isGeneratorVoltageRemoteControl() && controlledBus != controllerBus) {
+                    LOGGER.warn("Remote voltage control is not activated. The voltage target of {} with remote control is rescaled from {} to {}",
+                            controllerBus.getId(), controllerTargetV, controllerTargetV * controllerBus.getNominalV() / controlledBus.getNominalV());
+                } else if (parameters.getMaxVoltageRemoteControlDistance() != 0 && distanceBetweenBuses(controlledBus, controllerBus, parameters.getMaxVoltageRemoteControlDistance()) > parameters.getMaxVoltageRemoteControlDistance()) {
+                    LOGGER.error("Voltage controlled bus '{}' is too far from controller bus '{}' (maximum distance is {}). The bus switches to local voltage control and target voltage is rescaled from {} to {}",
+                            controlledBus.getId(), controllerBus.getId(), parameters.getMaxVoltageRemoteControlDistance(), controllerTargetV, controllerTargetV * controllerBus.getNominalV() / controlledBus.getNominalV());
+                    Reports.reportTooFarControlledBus(controlledBus.getNetwork().getReportNode(), controllerBus.getId(), controlledBus.getId(), parameters.getMaxVoltageRemoteControlDistance());
+                } else {
+                    keepVoltageRemoteControl = true;
+                }
+
+                if (keepVoltageRemoteControl) {
                     controlledBus.getGeneratorVoltageControl().ifPresentOrElse(
                         vc -> updateGeneratorVoltageControl(vc, controllerBus, controllerTargetV),
                         () -> createGeneratorVoltageControl(controlledBus, controllerBus, controllerTargetV, voltageControls, parameters));
                 } else {
                     // if voltage remote control deactivated and remote control, set local control instead
-                    LOGGER.warn("Remote voltage control is not activated. The voltage target of {} with remote control is rescaled from {} to {}",
-                            controllerBus.getId(), controllerTargetV, controllerTargetV * controllerBus.getNominalV() / controlledBus.getNominalV());
                     controlledBus.getGeneratorVoltageControl().ifPresentOrElse(
                         vc -> updateGeneratorVoltageControl(vc, controllerBus, controllerTargetV), // updating only to check targetV uniqueness
```

Comment:
What is the difference with the if case ? I do not understand how the voltage control is transformed to a local control in this case...